ICAFIX/scripts/run_icasso.m

```diff
@@ -0,0 +1,253 @@
+% It creates outputs in the style of MELODIC, so that FIX can be run
+% afterwards. Most but not all of MELODIC's outputs are created.
+% 
+% All of this function's inputs are strings
+%
+% Required Inputs (variable names verbatim from hcp_fix_multi_run.sh):
+%   Dim          : Data diminsionaliy estimate, typically Wishart-based
+%   concatfmri   : File name of original 4d time series without extension
+%   concatfmrihp : File name of high-passed 4d time series without extension
+%   ConcatFolder : Directory which contains concatfmri
+%
+% Optional Inputs:
+%   vis     : Whether to create and save icasso figures from last icasso level, see icasso.m 'basic' (default) or 'off'
+%   nICA    : Number of ICA repetitions per icasso repetition, @ delimited string, (default = '100')
+%   nClust  : Number of icasso repetitions (default = '1')
+%   maxIter : Maximum number of iterations per ica fit (default = '1000')
+%
+% example inputs:
+% Dim = '41';
+% concatfmri = 'tfMRI_EMOTION_RL_LR';
+% concatfmrihp = 'tfMRI_EMOTION_RL_LR_hp0';
+% ConcatFolder = '/mnt/myelin/burke/HCPpipelines/dev_study/100307/MNINonLinear/Results/tfMRI_EMOTION_RL_LR_ICASSO';
+% vis = 'basic';
+% nICA = '2@2';
+% nClust = '2';
+% maxIter = '1000';
+% 
+% Dim = str2double(Dim);
+% nICA = cellfun(@str2double,regexp(nICA,'@','split'));
+% nClust = str2double(nClust);
+% maxIter = str2double(maxIter);
+
+% Created 2024-06-05
+% Burke Rosen
+
+% ToDo:
+% () Currently the vnts file and brainMaskFile paths and names are inferred from
+%    ConcatFolder, concatfmrihp, and concatfmri. Maybe they should be their own 
+%    arguments for flexibility.
+
+%% parse parameters
+if nargin < 4 || any(cellfun(@isempty,{Dim,concatfmri,concatfmrihp,ConcatFolder}))
+  error('Dim, concatfmri, concatfmrihp, and ConcatFolder are required!')
+end
+Dim = str2double(Dim);
+if nargin < 5 || isempty(vis) 
+  vis = 'basic';
+end
+if ~ismember(vis,{'basic','off'})
+  warning('vis must be ''basic'' or ''off'', reverting to ''basic''')
+  vis = 'basic';
+end
+if nargin < 6 || isempty(nICA) 
+  nICA = 100;
+else
+  nICA = cellfun(@str2double,regexp(nICA,'@','split'));
+end
+if nargin < 7 || isempty(nClust) 
+  nClust = 1;
+else
+  nClust = str2double(nClust);
+end
+if numel(nICA) ~= nClust
+  error('nICA must be a @ delimited list with a number of elements equal to nClust')
+end
+if nargin < 8 || isempty(maxIter) 
+  maxIter = 1000;
+else
+  maxIter = str2double(maxIter);
+end
+
+%% check IO dependencies
+% use imaging processing toolbox utilities, or if those are not available use FSL utilities 
+function out = out2(fun);[~,out] = fun();end
+function out = out3(fun);[~,~,out] = fun();end
+if isempty(which('niftiread'))
+  infoNIFTI = @(fName) struct('ImageSize',out2(@() read_avw(fName))','PixelDimensions',out3(@() read_avw(fName))');
+  readNIFTI = @(fName) read_avw(fName);
+  writeNIFTI = @(img,fName,hdr) save_avw(img,fName,'f',hdr.PixelDimensions);
+  if isempty(which('read_avw'))
```

Comment:
Seems like it would be better to test for this before defining things that use the _avw functions.

Reply:
Fixed.